Tests/TuistScaffoldIntegrationTests/Utils/TemplatesDirectoryLocatorIntegrationTests.swift

```diff
@@ -0,0 +1,94 @@
+import Basic
+import Foundation
+import TuistSupport
+import XCTest
+
+@testable import TuistCore
+@testable import TuistScaffold
+@testable import TuistSupportTesting
+
+final class TemplatesDirectoryLocatorIntegrationTests: TuistTestCase {
+    var subject: TemplatesDirectoryLocator!
+
+    override func setUp() {
+        super.setUp()
+        RootDirectoryLocator.shared = RootDirectoryLocator()
```

Comment:
I'd rather extend the initializer of `TemplatesDirectoryLocator` and inject `rootDirectoryLocator` as a dependency:

```swift
TemplatesDirectoryLocator(rootDirectoryLocator:  RootDirectoryLocator())

class TemplatesDirectoryLocator {
  init(rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator.shared) {
    // Initialization
  }
}

```